server/modules/spotifyPlaylist.js

```diff
@@ -0,0 +1,81 @@
+var SpotifyWebApi = require('spotify-web-api-node');
+
+var scopes = ['playlist-read-private', 'playlist-modify-public', 'playlist-modify-private'],
+  clientId = "873379a32a354918b0a54f4f79736716",
+  clientSecret = process.env.CLIENT_SECRET,
+  state = 'jazz-and-eri-women-powahhh';
+
+var spotifyApi = new SpotifyWebApi({
+  clientId : clientId,
+  clientSecret : clientSecret
+});
+
+var authorizeURL = spotifyApi.createAuthorizeURL(scopes, state);
+
+exports.getTracksAndCreatePlaylist = function(arrayOfArtists, userId, accessToken, festivalName){
+  getTrackIDArray(arrayOfArtists)
+    .then(function(trackIDsArray){
+      newPlaylist(userId, accessToken, festivalName, trackIDsArray);
+    });
+}
+
+newPlaylist = function(username, accessTok, festivalName, trackIDsArray){
+  changeAccessToken(accessTok);
+  var playlistID = '';
+  createPlaylist(username, festivalName+' playlist')
+    .then(function(playlistID){
+      addTracksToPlaylist(username, playlistID, trackIDsArray)
+    })
+}
+
+changeAccessToken = function(accessTok){
+  spotifyApi.setAccessToken(accessTok);
+}
+
+getTrackIDArray = function(arrayOfArtists){
+  var trackIDsArray = [];
+  var promiseArray = [];
+  arrayOfArtists.forEach(function(artist){
+    var promise = searchForTracks(artist)
+      .then(function(response){
```

Comment:
Can some of this functionality be extracted to named functions, so that it reads more easily?